### ISSUE TITLE
Splunk does not log to systemout

### ISSUE THREAD

**Issue by lindsjan** [splunk/docker-splunk#231]
Running Splunk as HEC in Kubernetes. Works otherwise fine, except that occasionally my fluentd- daemonset gets 503 - errors from the HEC container, with message "Server is busy".  To my surprise, the Hec- container itself is not printing anything to systemout, that is to container log -  apart from the installation time Ansible rows. After that, no logging.. :

`
splunk-hec-86db94d796-292kw:splunk-hec PLAY RECAP *********************************************************************                                                                 
splunk-hec-86db94d796-292kw:splunk-hec localhost : ok=50   changed=12   unreachable=0    failed=0                                                                      │
 `
Shelling into the container, can see that splunkd.log is also empty.

What would be the best way to set up the logging so that can see from the container log what is wrong with Hec?   Did not find any log-related settings in https://splunk.github.io/splunk-ansible/advanced/default.yml.spec.html#spec

**Comment by nwang92** [comment]
You may have better luck by logging into Splunk and running an `index=_internal` search to see exactly what's happening. My guess is you may be overwhelming whatever default queue settings are defined for the HEC endpoint or you may be hitting open file handler limits.

But for your original ask, from https://splunk.github.io/splunk-ansible/ADVANCED.html#environment-variables-for-splunk-instance, there's a `SPLUNK_TAIL_FILE` env var which determines what logs you get from the container stdout/stderr. By default it's only configured for `splunkd_stderr.log` because that's the main process, but you can switch it to something else. The only problem in doing so is you'd need to know which logs you care about (as in this case, the logs of the individual HEC component are not in the same stderr log file) + output logging size (I've had to set some limits on the docker daemon's logging options from https://docs.docker.com/config/containers/logging/configure/)

**Comment by lindsjan** [comment]
thanks, checking the internal index. Seems there is something wrong with SPLUNK_TAIL_FILE.  Before tuning, there were rows in splunkd_stderr.log but none came to systemout.  After setting SPLUNK_TAIL_FILE to splunkd.log,  no difference, nothing in systemout.  

Maybe SPLUNK_TAIL_FILE is not obeyed. In container can see my value set:

SPLUNK_TAIL_FILE=splunkd.log

but ansible install still writes to systemout:

splunk-hec-6cc797cc7-bngrl:splunk-hec Ansible playbook complete, will begin streaming var/log/splunk/splunkd_stderr.log

**Comment by nwang92** [comment]
I believe you'll need to pass in the full file path, something like:
```
$ docker run ... -e SPLUNK_TAIL_FILE=/opt/splunk/var/log/splunk/splunkd.log splunk/splunk:latest
```

Or for multiple files and/or the universal forwarder image, you can refer to the env vars we have for the tests over this:
https://github.com/splunk/docker-splunk/blob/develop/tests/test_docker_splunk.py#L690
https://github.com/splunk/docker-splunk/blob/develop/tests/test_docker_splunk.py#L726

**Comment by lindsjan** [comment]
Strange, the env variable is set correctly...

ansible@splunk-hec:/$ printenv | grep TAIL
SPLUNK_TAIL_FILE=/opt/splunk/var/log/splunk/splunkd.log

..but when looking at containers processes, can see there is tail with command line
ansible@splunk-hec:/$ cat /proc/1263/cmdline
tail-n0-f/opt/splunk/var/log/splunk/splunkd_stderr.log

so its not obeying entrypoint.sh ?

**Comment by nwang92** [comment]
Are you using the `splunk/splunk` image or the `splunk/universalforwarder` image? I'm looking back on the timeline of this change, and it looks like this `SPLUNK_TAIL_FILE` feature was added April of this year. So this should be available in the >=7.2.7 tags of the above images.

**Comment by lindsjan** [comment]
Thanks, that explained - had splunk:7.2.5.1 in use. Now after updating works fine!
